### ISSUE TITLE
useValue bugged in latest version

### ISSUE THREAD

**Issue by mambax** [nestjs/nest#3045]
## Current behavior
I try to write a dynamic module. Check the gist from 🙏 @evolkmann. I get a compiler error:

`Returned expression type ... is not assigneable to DynamicModule`

If you switch to `useClass` it seamlessly works. What am I doing wrong or is it a bug?

## Input Code
https://gist.github.com/evolkmann/1019721d46304df3d8ff7adbca87d8ec

## Expected behavior
No compiler error since this should work acc. to official docs and I assume gist worked once.

## Possible Solution
It works with 
```
{
 provide: MyLibService,
  useFactory: () => {
  return new MyLibService(type);
 },
}
```

But it should accept a value too...?

## Environment

<pre><code>
"@nestjs/common": "^6.7.2",
"@nestjs/core": "^6.7.2",
 
For Tooling issues:
- Node version:
"node": "10.16.2"
"npm": "6.10.3"
- Platform: Mac

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by evolkmann** [comment]
@mambax maybe you can find more clarification from [the blog post](https://medium.com/@enzo_volkmann/how-to-pass-configuration-into-your-nest-js-modules-1f883faca4f7) referencing the gist you mentioned. I don't know if you read it before coming to the gist.

As @kamilmysliwiec mentioned, it is hard to find your specific problem without an explicit example.

Maybe provide some code and I can try to help.

**Comment by johnbiundo** [comment]
@mambax you might find [this blog post](https://dev.to/nestjs/advanced-nestjs-how-to-build-completely-dynamic-nestjs-modules-1370) helpful.

**Comment by mambax** [comment]
Hey all
I learned from your posts, I read them before, here is the repo:
https://github.com/mambax/use-value-bug

![image](https://user-images.githubusercontent.com/5472171/65749058-a58bcf00-e105-11e9-9bbb-a1b69b852bd0.png)
I am still not excluding a layer 8 problem (me) I just don't get what could be wrong :(

**Comment by evolkmann** [comment]
Hi @mambax, for me your code works fine. The only thing which has to be changed is the app module. You have to call the `register()` method there.

```
import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { AuthModule } from './auth/auth.module';

@Module({
  imports: [
    AuthModule.register({
      type: 'jwt'
    })
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}
```

**Comment by mambax** [comment]
@evolkmann Thanks for checking. I wonder what is wrong with my IDE then. The register call from you of course is missing, but will it solve the typing error above?

**Comment by evolkmann** [comment]
@mambax I don't know, for me there are no errors, with and without the `register()`-call. I installed the exact dependencies defined in the `package-lock.json`. I am using VS Code, but IntelliJ should also handle it of course.

**Comment by mambax** [comment]
@evolkmann thanks for helping out anyway, I don't get what is wrong with that thing 😢

**Comment by mambax** [comment]
🎺 
Turns out, `rm -rf .idea` does not completely invalidate all caches, weird! I also deleted caches `File > Invalidate caches / Restart` and now it works. Sorry everyone to bother.

**Comment by angelov-todor** [comment]
this change for me in the new version 6.7.2.
```
GraphQLModule.forRootAsync({
      imports: [LoggerModule],
      inject: [LoggerService],
      useFactory: (logger: LoggerService) => {
        return new GqlConfigService(logger).createGqlOptions();
      }
    }),
```
to 
```
GraphQLModule.forRootAsync({
      imports: [SVLoggerModule],
      inject: [SvLoggerService],
      useFactory: (logger: SvLoggerService) => {
        return new GqlConfigService(logger[0]).createGqlOptions();
      }
    }),
```
somehow the parameter is injected as an array ?

**Comment by dboskovic** [comment]
Can confirm useFactory has stopped working for us where inject is an array.

**Comment by kamilmysliwiec** [comment]
@angelov-todor @dboskovic just fixed in `@nestjs/graphql@6.5.3`. Thanks for reporting!
